consensus/src/quorum_store/batch_generator.rs

```diff
@@ -217,22 +219,65 @@ impl BatchGenerator {
         batches
     }
 
+    fn insert_batch_in_progress(
+        &mut self,
+        batch_id: BatchId,
+        txns_in_progress: Vec<(TransactionSummary, TransactionInProgress)>,
+    ) {
+        let mut txns = vec![];
+        for (summary, info) in txns_in_progress {
+            let txn_info = self
+                .txns_in_progress_sorted
+                .entry(summary)
+                .or_insert_with(|| TransactionInProgress::new(info.gas_unit_price));
+            txn_info.increment();
+            txn_info.gas_unit_price = info.gas_unit_price.max(txn_info.gas_unit_price);
```

Comment:
should we overwrite with higher gas price? i.e. don't increment and decrement if gas unit price not equal